open-api/rest-catalog-open-api.yaml

```diff
@@ -2106,6 +2213,32 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        An opaque JSON object that contains information provided by the REST server,
+        to be utilized by clients for distributed table scan planning, should be supplied
+        as is for input in `PlanTable` operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - data-file
+      properties:
```

Comment:
this is missing a bunch of other properties. See also `FileScanTaskParser`

Reply:
Originally I had included all the required properties from the `FileScanTaskParser` https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/FileScanTaskParser.java#L33. 

However after discussing on the PR with @rdblue @jackye1995 it seems we are not modeling the `FileScanTaskParser` one to one with the spec. Instead we are trying to opt for something more concise, see the following comment here https://github.com/apache/iceberg/pull/9695#discussion_r1494924550 

After several revisions it seems the only required property we wanted for `FileScanTask` was `data-file`. Feel free to correct me if my understanding is incorrect @rdblue @jackye1995